### PR TITLE
feat(avatar): added size variations

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/4647

--

this also adds back a missing page demo template file that was throwing hbs errors on a rebase.

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
Thanks for doing this so quickly @mcoker . Now that I'm seeing this, I wonder if the small avatar is really useful as I don't think you can fit a recognizable image there.

@mceledonia what if we got rid of the 16px variant and made 24px the smallest and bumped everything else. So XL would then be 128px?

**Review by srambach — APPROVED** [review]
LGTM 👍

File: src/patternfly/components/Avatar/avatar.scss [hunk 1]
```diff
@@ -27,4 +40,11 @@
     --pf-c-avatar--BorderColor: var(--pf-c-avatar--m-dark--BorderColor);
     --pf-c-avatar--BorderWidth: var(--pf-c-avatar--m-dark--BorderWidth);
   }
+
+  @each $size in $sizes {
```
Comment (mattnolting) [inline src/patternfly/components/Avatar/avatar.scss:44]: 😎

**Review by mattnolting — APPROVED** [review]
L🍕TM

File: src/patternfly/components/Avatar/examples/Avatar.md [hunk 2]
```diff
@@ -22,6 +22,31 @@ import './Avatar.css'
...
+### Extra large
+```hbs
+{{> avatar avatar--modifier="pf-m-xl" avatar--attribute='src="/assets/images/img_avatar-light.svg" alt="Avatar image extra large"'}}
+```
+
+### Extra extra large
+```hbs
+{{> avatar avatar--modifier="pf-m-2xl" avatar--attribute='src="/assets/images/img_avatar-light.svg" alt="Avatar image extra extra large"'}}
+```
+
```
Comment (srambach) [inline src/patternfly/components/Avatar/examples/Avatar.md:49]: I think this isn't needed anymore?

Comment (mcoker) [reply src/patternfly/components/Avatar/examples/Avatar.md]: updated! thanks for spotting that.

File: src/patternfly/components/Avatar/examples/Avatar.md [hunk 3]
```diff
@@ -37,3 +57,7 @@ The avatar component provides a default SVG icon. If an image is used it should
 | `.pf-c-avatar` | `<img>` |  Initiates an avatar image. **Required** |
 | `.pf-m-light` | `.pf-c-avatar` | Modifies an avatar for use against a light background. |
 | `.pf-m-dark` | `.pf-c-avatar` | Modifies an avatar for use against a dark background. |
+| `.pf-m-sm` | `.pf-c-avatar` | Modifies an avatar to be small. |
+| `.pf-m-md` | `.pf-c-avatar` | Modifies an avatar to be medium. |
+| `.pf-m-lg` | `.pf-c-avatar` | Modifies an avatar to be large. **Note:** This is the default size. |
+| `.pf-m-xl` | `.pf-c-avatar` | Modifies an avatar to be extra large. |
```
Comment (srambach) [inline src/patternfly/components/Avatar/examples/Avatar.md:63]: Should we include "at the optional breakpoint" since we're generating all those classes?

**Comment by mcarrano** [comment]
This looks great @mcoker . My only issue is that you say in the documentation that `pf-m-lg` is the default size. But it looks like medium is actually the default size. Maybe just an artifact of our bumping up the sizes?

File: src/patternfly/components/Avatar/examples/Avatar.md (hunk 3 above)
Comment (mcoker) [reply src/patternfly/components/Avatar/examples/Avatar.md]: yep! updated with your new link.

**Comment by mcoker** [comment]
@mcarrano nice catch, updated!

**Review by srambach — APPROVED** [review]
🎉

**Review by mcarrano — APPROVED** [review]
Looks great!

**Pull request merged**